cmd/create.go

```diff
@@ -286,3 +288,41 @@ func filterYaml(cmd *cobra.Command, args []string, toComplete string) ([]string,
 	}
 	return []string{"yaml", "yml"}, cobra.ShellCompDirectiveFilterFileExt
 }
+
+// Determines if overwrite is allowed
+func checkOverwriteAllowed(depDir string, bp config.Blueprint, overwriteFlag bool) error {
+	if _, err := os.Stat(depDir); os.IsNotExist(err) {
+		return nil // all good, no previous deployment
+	}
+
+	useWErr := fmt.Errorf("deployment folder %q already exists, use -w to overwrite", depDir)
+	// try to get previous deployment
+	expPath := filepath.Join(modulewriter.ArtifactsDir(depDir), modulewriter.ExpandedBlueprintName)
+	prev, _, err := config.NewDeploymentConfig(expPath)
+	if err != nil { // malformed previous deployment, rely on flag
+		if overwriteFlag {
+			return nil
+		}
+		return useWErr
```

Comment:
I think would prefer to just not allow overwrite in the malformed case. Can we tell the user to use an intermediate version to bridge the gap, or just stop them. In this case we know they are re-applying with a changing ghpc version which we specifically do not recommend. Maybe instead of version warning, in this case we do version error "you are trying to change version of ghpc and your previous version is too old"